feat(get_trt): Allow estimated and fallback TotalReadoutTime

effigies opened the pull request:
This is work toward https://github.com/nipreps/fmriprep/issues/3009.

The primary purpose is to recover SyN-SDC in cases where TRT is not calculable. When we calculated displacement fields and applied them without round-tripping through a field in Hz, this didn't matter. Now that we do, it would be good to allow tools to supply a trivial value in these cases. I would suggest a power of 2 that is in the neighborhood of a plausible value, such as 0.03125, which only modifies the exponent of a floating point value, and not the significand, minimizing the potential for accumulating floating point error.

While I'm here, I provide a boolean flag (`False` by default) to fall back to EstimatedTotalReadoutTime or EstimatedEffectiveEchoSpacing.

The order of precedence (if both `use_estimated` and `fallback` are enabled) is:

1) TotalReadoutTime
2) EstimatedTotalReadoutTime
3) EffectiveEchoSpacing
4) EstimatedEffectiveEchoSpacing
5) EchoSpacing + ParallelReductionFactorInPlane
6) WaterFatShift + EPIFactor
7) fallback

We could instead move `Estimated*` to after `WaterFatShift`, if that would be better.

Closes #249.

effigies committed:
feat(get_trt): Allow estimated and fallback TotalReadoutTime

oesteban commented on sdcflows/interfaces/epi.py:
```diff
@@ -37,6 +37,8 @@
 class _GetReadoutTimeInputSpec(BaseInterfaceInputSpec):
     in_file = File(exists=True, desc="EPI image corresponding to the metadata")
     metadata = traits.Dict(mandatory=True, desc="metadata corresponding to the inputs")
+    use_estimate = traits.Bool(False, desc='Use "Estimated*" fields to calculate TotalReadoutTime')
```
If I understand Chris Rorden's thinking, this perhaps is best if enabled? Also, `usedefault` is missing

```suggestion
    use_estimate = traits.Bool(True, usedefault=True, desc='Use "Estimated*" fields to calculate TotalReadoutTime')
```

effigies replied:
My thought was that since this isn't BIDS, using these fields should be opt-in instead of default.

Edit: `Estimated*` adds some friction where the user needs to take affirmative action to use these estimates, so I see this approach as respecting that intention. I'm not wedded to this position, though, so I'm okay being overruled.

effigies replied:
```suggestion
    use_estimate = traits.Bool(False, usedefault=True, desc='Use "Estimated*" fields to calculate TotalReadoutTime')
```

effigies committed:
fix: Check for PhaseEncodingDirection

We previously allowed a KeyError to be raised, but that makes falling
back more difficult. This puts the entire PEDir-dependent branch inside
an if-block.

effigies committed:
Update sdcflows/interfaces/epi.py

effigies committed:
sty: Fix style complaint
